client/client-multi/client-objective-c/Kaa/DefaultEventManager.m

```diff
@@ -0,0 +1,263 @@
+        @synchronized(self.eventGuard) {
+            Event *event = [[Event alloc] init];
+            event.seqNum = [self.state getAndIncrementEventSequenceNumber];
+            event.eventClassFQN = eventFQN;
+            event.eventData = [NSData dataWithData:data];
+            if (target) {
+                event.target = [KAAUnion unionWithBranch:KAA_UNION_STRING_OR_NULL_BRANCH_0 andData:target];
+            }
+            [self.currentEvents addObject:event];
+        }
+        
+        if (!self.isEngaged) {
+            [self.transport sync];
+        }
+    }
+}
+
+- (void)registerEventFamily:(id<BaseEventFamily>)eventFamily {
+    [self.registeredEventFamilies addObject:eventFamily];
+}
+
+- (void)onGenericEvent:(NSString *)eventFQN data:(NSData *)data source:(NSString *)source {
+    DDLogInfo(@"%@ Received event [eventClassFQN: %@]", TAG, eventFQN);
+    for (id<BaseEventFamily> family in self.registeredEventFamilies) {
+        DDLogInfo(@"%@ Lookup event fqn %@ in family %@", TAG, eventFQN, family);
+        if ([[family getSupportedEventFQNs] containsObject:eventFQN]) {
+            DDLogInfo(@"%@ Event fqn [%@] found in family [%@]", TAG, eventFQN, family);
+            [[self.executorContext getCallbackExecutor] addOperationWithBlock:^{
+                [family onGenericEvent:eventFQN withData:data from:source];
+            }];
+        }
+    }
+}
+
+- (NSInteger)findEventListeners:(NSArray *)eventFQNs delegate:(id<FindEventListenersDelegate>)delegate {
+    int requestId = arc4random();
+    EventListenersRequest *request = [[EventListenersRequest alloc] init];
+    request.requestId = requestId;
+    request.eventClassFQNs = eventFQNs;
+    EventListenersRequestBinding *bind = [[EventListenersRequestBinding alloc] initWithRequest:request delegate:delegate];
+    [self.eventListenersRequests setObject:bind forKey:[NSNumber numberWithInt:requestId]];
+    DDLogDebug(@"%@ Adding event listener resolution request. Request ID: %i", TAG, requestId);
+    if (!self.isEngaged) {
+        [self.transport sync];
+    }
+    return requestId;
+}
+
+- (void)eventListenersResponseReceived:(NSArray *)response {
+    for (EventListenersResponse *singleResponse in response) {
+        DDLogDebug(@"%@ Received event listener resolution response: %@", TAG, singleResponse);
+        NSNumber *key = [NSNumber numberWithInt:singleResponse.requestId];
+        __block EventListenersRequestBinding *bind = [self.eventListenersRequests objectForKey:key];
+        if (bind) {
+            [self.eventListenersRequests removeObjectForKey:[NSNumber numberWithInt:singleResponse.requestId]];
+            [[self.executorContext getCallbackExecutor] addOperationWithBlock:^{
+                if (singleResponse.result == SYNC_RESPONSE_RESULT_TYPE_SUCCESS
+                    && singleResponse.listeners.branch == KAA_UNION_ARRAY_STRING_OR_NULL_BRANCH_0) {
```

Comment:
Call the "success" callback disregarding whether array of listeners is received or not.